lib/util/Arrays.v3

```diff
@@ -19,6 +19,17 @@ component Arrays {
 		for (i < x.length) if (x[i] != y[i]) return false;
 		return true;
 	}
+	// Compare string elements of {x} and {y} for equality.
+	def strEqual(x: Array<string>, y: Array<string>) -> bool {
+		return equal_with(x, y, Strings.equal);
+	}
+	// Check equality of {x} and {y} with custom equality function.
+	def equal_with<A>(x: Array<A>, y: Array<A>, eq: (A, A) -> bool) -> bool {
+		if (x == y) return true;
+		if (x.length != y.length) return false;
+		for (i < x.length) if (!eq(x[i], y[i])) return false;
+		return true;
+	}
```

Comment:
It'd be better to refactor `equal` to use this `equal_with` passing in a closure...but I don't know if there's an inline way to define a closure? e.g.

```
def equal<A>(x: Array<A>, y: Array<A>) -> bool {
  return equal_with(x, y, |a, b| -> bool {
    a == b;
  });
}
```

Reply:
I guess that'd just be the function that handles the `==` operation...I'll check to see if that's available somewhere to pass.

Reply:
Oh daaaang...that's niiiiice

`return equal_with(x, y, A.==);`